### PR TITLE
Add NodeStatusRecorder

### PR DESCRIPTION
**Pull request by mrtracy**
NodeStatusRecorder is designed to convert the current status in a
NodeStatusMonitor into Time Series data.

My current plan is for the time series package to periodically call
NodeStatusRecorder.GetTimeSeriesData() and persist the returned values; this
will be facilitated with an interface added to the time series package.

### CODE REVIEW

**Comment by bdarnell** [comment]
LGTM

File: server/status/recorder.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+// storeStatusRecorder is a helper class for recording time series datapoints
+// from a single StoreStatusMonitor.
+type storeStatusRecorder struct {
+	*StoreStatusMonitor
+	timestampNanos int64
+}
+
+// recordInt records a single int value from the StoreStatusMonitor as a
+// proto.TimeSeriesData object.
+func (ssr *storeStatusRecorder) recordInt(name string, data int64) proto.TimeSeriesData {
```
Comment (BramGruneir) [inline server/status/recorder.go]: This is named recordInt, but the next function is intDatapoint.  Perhaps either prepend or postpend the int.  Are you planning on having other types?

File: server/status/recorder_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+func (a byTimeAndName) Len() int      { return len(a) }
+func (a byTimeAndName) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byTimeAndName) Less(i, j int) bool {
+	if a[i].Name != a[j].Name {
+		return a[i].Name < a[j].Name
+	}
+	return a[i].Datapoints[0].TimestampNanos < a[j].Datapoints[0].TimestampNanos
+}
+
+func TestNodeStatusRecorder(t *testing.T) {
```
Comment (BramGruneir) [inline server/status/recorder_test.go]: Please describe what this test is doing.

**Comment by BramGruneir** [comment]
LGTM

**Comment by spencerkimball** [comment]
I think you probably want this recorder to call the ts package and not the other way around. The ts package is lower level. This should use it. Or did you just misspeak in your commit description?

File: server/status/recorder_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+func (a byTimeAndName) Less(i, j int) bool {
+	if a[i].Name != a[j].Name {
+		return a[i].Name < a[j].Name
+	}
+	return a[i].Datapoints[0].TimestampNanos < a[j].Datapoints[0].TimestampNanos
+}
+
+// TestNodeStatusRecorder verifies that the time series data generated by a
+// recorder matches the data added to the monitor.
+func TestNodeStatusRecorder(t *testing.T) {
```
Comment (spencerkimball) [inline server/status/recorder_test.go]: Good test.

File: server/status/recorder.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+// GetTimeSeriesData returns a slice of interesting TimeSeriesData from the
+// encapsulated NodeStatusMonitor.
+func (nsr *NodeStatusRecorder) GetTimeSeriesData() []proto.TimeSeriesData {
+	data := make([]proto.TimeSeriesData, 0, nsr.lastDataCount)
+	nsr.VisitStoreMonitors(func(ssm *StoreStatusMonitor) {
+		now := nsr.clock.PhysicalNow()
+		ssr := storeStatusRecorder{ssm, now}
+		data = append(data, ssr.recordInt("livebytes", ssr.stats.LiveBytes))
```
Comment (spencerkimball) [inline server/status/recorder.go]: It's confusing to use `ssr.stats` here. Why not just use `ssm.stats`?

**Comment by spencerkimball** [comment]
LGTM

**Pull request merged**